radiosity.cpp

```diff
@@ -536,41 +487,34 @@ int main(int argc, char const* argv[]) {
     KDTree tree(std::move(triangles));
 
     // Image
-    int width = args["--width"].asLong();
+    int width = conf.width;
     assert(width > 0);
     int height = width / cam.mAspect;
     Image image(width, height);
 
     // Compute radiosity
     std::vector<Color> radiosity;
 
-    if (args["direct"].asBool()) {
+    // TODO: split in functions
+    if (conf.mode == RadiosityConfig::EXACT) {
         radiosity = compute_radiosity(tree);
         image = raycast(tree, conf, cam, radiosity, std::move(image));
-        if (args["--rad-simple-mesh"] && args["--rad-simple-mesh"].asBool()) {
+        if (conf.mesh == RadiosityConfig::SIMPLE_MESH) {
             image = render_mesh(tree.triangles(), cam, std::move(image));
-        } else {
+        } else if (conf.mesh == RadiosityConfig::FEATURE_MESH) {
             image = render_feature_lines(tree, conf, cam, std::move(image));
         }
-    } else if (args["hierarchical"].asBool()) {
-        // compute minimal area
-        int max_subdivisions = args["--max-subdivisions"].asLong();
-        float min_area = ::min(tree.triangles().begin(), tree.triangles().end(),
-                               [](const Triangle& tri) { return tri.area(); });
-        min_area /= pow(4, max_subdivisions);
-        std::cerr << "Minimal area: " << min_area << std::endl;
-
-        float F_eps = std::stof(args["--form-factor-eps"].asString());
-        std::cerr << "Form factor epsilon: " << F_eps << std::endl;
-
-        size_t max_iterations = args["--max-iterations"].asLong();
-        std::cerr << "Maximum iterations: " << max_iterations << std::endl;
-
-        float BF_eps = std::stof(args["--rad-shoot-eps"].asString());
-        std::cerr << "Shooting radiosity epsilon: " << BF_eps << std::endl;
-
-        HierarchicalRadiosity model(tree, F_eps, min_area, BF_eps,
-                                    max_iterations);
+    } else if (conf.mode == RadiosityConfig::HIERARCHICAL) {
+        conf.min_area = ::min(tree.triangles().begin(), tree.triangles().end(),
+                              [](const Triangle& tri) { return tri.area(); });
+        conf.min_area /= pow(4, conf.max_subdivisions);
+        std::cerr << "Minimal area: " << conf.min_area << std::endl;
+        std::cerr << "Form factor epsilon: " << conf.F_eps << std::endl;
+        std::cerr << "Maximum iterations: " << conf.max_iterations << std::endl;
+        std::cerr << "Shooting radiosity epsilon: " << conf.BF_eps << std::endl;
```

Comment:
We could, and then output it a verbose flag is specified. I would say this is a subject for the pull request. :)